py/qqa/qa/snr.py

```diff
@@ -0,0 +1,182 @@
+from .base import QA
+import glob
+import os
+import collections
+
+import numpy as np
+import fitsio
+
+from astropy.table import Table
+from astropy import units,constants
+
+import desiutil.log
+from desispec.qproc.io import read_qframe
+from desispec.calibfinder import CalibFinder
+from desispec.interpolation import resample_flux
+from desispec.io.filters import load_filter
+from desispec.io.fluxcalibration import read_average_flux_calibration
+
+from desimodel.io import load_desiparams
+
+from desitarget import targets
+
+
+class QASNR(QA):
+    """docstring """
+    def __init__(self):
+        self.output_type = "PER_FIBER"
+    
+        # load things we will use several times
+        desiparams = load_desiparams()
+        
+        geometric_area_cm2 = 1e4 * desiparams["area"]["geometric_area"] # m2
+        
+        self.thru_conversion_wavelength = 6000 # A
+        energy_per_photon = (constants.h * constants.c / (self.thru_conversion_wavelength*units.Angstrom)).to(units.erg).value #ergs
+        self.thru_conversion_factor_ergs_per_cm2 = 1e17*energy_per_photon/geometric_area_cm2 # ergs/cm2 
+        
+
+    def valid_flavor(self, flavor):
+        return ( flavor.upper() == "SCIENCE" )
+
+    def run(self, indir):
+        '''TODO: document'''
+
+        log = desiutil.log.get_logger()
+
+        results = list()
+
+        infiles = glob.glob(os.path.join(indir, 'qcframe-*.fits'))
+        if len(infiles) == 0 :
+            log.error("no qcframe in {}".format(indir))
+            return None
+
+        
+
+        filters = None
+
+        # find number of spectros
+        spectros=[]
+        for filename in infiles:
+            hdr=fitsio.read_header(filename)
+            s=int(hdr['CAMERA'][1])
+            spectros.append(s)
+        spectros=np.unique(spectros)
+        
+        for spectro in spectros :
+            
+            infiles = glob.glob(os.path.join(indir, 'qcframe-?{}-*.fits'.format(spectro)))
+                        
+            qframes={}
+            fmap=None
+            for infile in infiles :
+                qframe=read_qframe(infile)
+                cam=qframe.meta["CAMERA"][0].upper()
+                qframes[cam]=qframe
+                
+                if fmap is None :
+                    fmap = qframe.fibermap
+                    night = int(qframe.meta['NIGHT'])
+                    expid = int(qframe.meta['EXPID'])
+                    
+                    if filters is None : # load filters
+                        filters = dict()
+                        
+                        if 'S' in fmap['PHOTSYS']:
+                            for b in ["G","R","Z"] :
+                                filters[b] = load_filter("DECAM_"+b)
+                                
+                        if 'N' in fmap['PHOTSYS']:
+                            for b in ["G","R","Z"] :
+                                filters[b] = load_filter("BASS_"+b)
```

Comment:
For tiles that span the N/S boundary, they will have both 'S' and 'N' in their PHOTSYS list, but this code will overwrite the DECAM filters with BASS.  Additionally, desispec.io.load_filter doesn't seem to know about the northern filters.  You probably also mean MZLS_Z, not BASS_Z?  speclite.filters.load_filter has these filters under the names BASS-G, BASS-R, and MzLS-R (note: dash not underscore, and the z in MzLS is lowercase).

I wouldn't mind if desispec was updated to support desispec.io.load_filters('G', photsys='N'), desispec.io.load_filters('Z', photsys='S'), etc.

For that matter, it looks like desispec.scripts.stdstars assumes that load_filter('BASS_G') works, but it apparently doesn't, though I'm surprised we haven't tested that.  It does try 'MZLS_Z' not 'BASS_Z' though.